packages/web/pages/stake/index.tsx

```diff
@@ -71,33 +85,94 @@ export const Staking: React.FC = observer(() => {
     return { currency: osmo, amount: primitiveAmount.amount, denom: osmo };
   }, [osmo, primitiveAmount]);
 
+  const delegationQuery = cosmosQueries.queryDelegations.getQueryBech32Address(
+    account?.address ?? ""
+  );
+
+  const userValidatorDelegations = delegationQuery.delegations;
+
+  const usersValidatorsMap = useMemo(() => {
+    const delegationsMap = new Map<string, StakingType.Delegation>();
+
+    userValidatorDelegations.forEach((delegation: StakingType.Delegation) => {
+      delegationsMap.set(delegation.delegation.validator_address, delegation);
+    });
+
+    return delegationsMap;
+  }, [userValidatorDelegations]);
+
+  const [amountDefault, setAmountDefault] = useState<AmountDefault>("input");
+
+  const getProperties = useCallback(() => {
+    const amountUSD = priceStore
+      .calculatePrice(
+        new CoinPretty(
+          osmo,
+          new Dec(amountConfig.amount).mul(
+            DecUtils.getTenExponentNInPrecisionRange(osmo.coinDecimals)
+          )
+        )
+      )
+      ?.toString();
+
+    const amount = amountConfig.amount;
+
+    const properties: Properties = {
+      amountDefault,
+      amount,
+      amountUSD,
+    };
+
+    return properties;
+  }, [amountConfig.amount, amountDefault, osmo, priceStore]);
```

Comment:
A few things I notice:
* It appears the stake event requires a lot of property data, which is fine, but can cause issues depending on how our code is organized.
* The fact that we have to query so much data at this top level, perhaps is a sign we're logging in the wrong layer? The wrong component? What if we log the stake event closer to where the data is being displayed? Querying at this top level will cause a lot of cascading renders as data loads, which is not a huge issue.